game/models.py

```diff
@@ -1,65 +1,48 @@
 
-class Competition(models.Model):
-    name = models.CharField(verbose_name=_('Name'), max_length=100)
-    server = models.ForeignKey(to='game.DockerContainer', verbose_name=_('Server'), null=False, blank=False)
-    additional_containers = models.ManyToManyField('game.DockerContainer', verbose_name=_("additional containers"),
-                                                   related_name='+', blank=True)
+
+__all__ = ["Game", "Operation", "OperationResource", "OperationParameter"]
+
 
 class Game(models.Model):
-    STATUS_VALUES = (
-        (0, _('Queued')),
-        (1, _('Running')),
-        (2, _('Finished')),
-        (3, _('Failed')),
-    )
+    name = models.CharField(verbose_name=_('name'), max_length=100)
 
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    competition = models.ForeignKey(to='game.Competition', verbose_name=_('Competition'), null=False, blank=False)
-    config = models.ForeignKey(to='storage.File', verbose_name=_('Game Configuration'), null=False, blank=False)
-    status = models.PositiveSmallIntegerField(verbose_name=_('status'), choices=STATUS_VALUES, default=0)
-    start = models.DateTimeField(verbose_name=_('Started At'), auto_now=True)
-    end = models.DateTimeField(verbose_name=_('Ended At'), auto_now=True)
-    time_limit = models.IntegerField(verbose_name=_('Time Limit (s)'))
 
-    def __unicode__(self):
-        return self.name
+class Operation(models.Model):
+    game = models.ForeignKey(to=Game, verbose_name=_('game'), null=False, blank=False)
+    name = models.CharField(verbose_name=_('name'), max_length=100)
+    config = models.TextField(verbose_name=_("docker compose yml template"), null=False)
 
-class Team(models.Model):
-    name = models.CharField(verbose_name=_('Name'), max_length=50)
-    game = models.ForeignKey(to='game.Game', verbose_name=_('Game'), related_name='+')
-    code = models.ForeignKey(to='storage.File', verbose_name=_('Code'), related_name='+')
-    data = models.ForeignKey(to='storage.File', verbose_name=_('Data'), related_name='+')
-    lang = models.ForeignKey(to='game.CompetitionLanguage', verbose_name=_('Language'))
+    class Meta:
+        unique_together = (("game", "name"), )
 
     def __unicode__(self):
         return self.name
 
-class CompetitionLanguage(models.Model):
-    name = models.CharField(verbose_name=_('title'), max_length=200)
-    competition = models.ForeignKey(to='game.Competition', verbose_name=_('Competition'), null=False, blank=False)
-    compile_container = models.ForeignKey('game.DockerContainer', verbose_name=_('Compile Container'), related_name='+',
-                                          null=True, blank=True)
-    execute_container = models.ForeignKey('game.DockerContainer', verbose_name=_('Execute Container'), related_name='+',
-                                          null=True, blank=True)
 
-    def __unicode__(self):
-        return self.name
+class OperationResource(models.Model):
```

Comment:
No. To define an operation some additional files might be required. These are kinda like operation static files. One might reference them in the Docker-Compose template file.